### ISSUE TITLE
PDO::fetchColumn return types seem wrong

### ISSUE THREAD

**Issue by NicolasCARPi** [vimeo/psalm#1632]
Hello,

Psalm expects PDO::fetchColumn to have a return type of string|null|false.

But both the [documentation](https://www.php.net/manual/en/pdostatement.fetchcolumn.php) and the [source code](https://github.com/php/php-src/blob/master/ext/pdo/pdo_stmt.c#L1361) seem to indicate that the return type can be only string|false.

This leads to false positive errors.

If this is a bug, I'll be happy to make a small PR addressing it :)

Cheers,
~Nico

**Comment by weirdan** [comment]
```php
var_dump($conn->query('select null from dual')->fetchColumn(0));
```
does return `null`.

**Comment by weirdan** [comment]
And with emulated prepares off you can get other types as well:
```php
$conn->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
var_dump($conn->query('select 1 from dual')->fetchColumn(0)); // int(1)
var_dump($conn->query('select "2" from dual')->fetchColumn(0)); // string(1) "2"
var_dump($conn->query('select atan(3) from dual')->fetchColumn(0)); // double(1.2490457723983)
var_dump($conn->query('select null from dual')->fetchColumn(0)); // NULL
```

**Comment by NicolasCARPi** [comment]
@weirdan Ok that's good to know, thanks! :)
